lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -1,17 +1,50 @@
+import 'dart:async';
+import 'package:pokedex_async_redux/features/pokemon_overview/widgets/circular_progress_indicator.dart';
 import 'package:pokedex_async_redux/features/pokemon_overview/widgets/pokemon_card.dart';
 import 'package:pokedex_async_redux/utilities/async.dart';
 import 'package:pokedex_async_redux/utilities/colors.dart';
 import 'package:pokedex_async_redux/utilities/constant.dart';
 import 'package:pokedex_async_redux/api/model/pokemon.dart';
 import 'package:flutter/material.dart';
 
-class PokemonOverviewPage extends StatelessWidget {
+class PokemonOverviewPage extends StatefulWidget {
   const PokemonOverviewPage({
     required this.pokemons,
+    required this.searchedPokemons,
+    required this.onSearchedPokemons,
+    required this.onClearSearchedPokemons,
     Key? key,
   }) : super(key: key);
 
   final Async<List<Pokemon>> pokemons;
+  final List<Pokemon> searchedPokemons;
+  final ValueChanged onSearchedPokemons;
+  final VoidCallback onClearSearchedPokemons;
+
+  @override
+  State<PokemonOverviewPage> createState() => _PokemonOverviewPageState();
+}
+
+class _PokemonOverviewPageState extends State<PokemonOverviewPage> {
+  late final TextEditingController _searchController;
+  Timer? _debounce;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _searchController = TextEditingController()..addListener(_onSearchChanged);
+  }
+
+  @override
+  void dispose() {
+    _debounce?.cancel();
+    _searchController.dispose();
+    _searchController.removeListener(_onSearchChanged);
```

Comment:
no need for this, since you already called dispose() above then everything most likely are cleared already 🙂